Examples/restcomm-olympus-nopod/restcomm-olympus/MainTableViewController.mm

```diff
@@ -131,33 +131,32 @@ - (void)viewDidLoad {
     
     [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(register:) name:UIApplicationDidBecomeActiveNotification object:nil];
     [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(unregister:) name:UIApplicationDidEnterBackgroundNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(appWillEnterForeground:) name:UIApplicationWillEnterForegroundNotification object:nil];
     
     self.contactsStore = [[CNContactStore alloc] init];
 
     //define spinner
     self.spinner = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
-    
-    //get contacts from phone
+    //get contacts first time
     [self checkContactsAccess];
```

Comment:
Isn't this redundant, now that you introduced it below?

Reply:
No,
- whenV viewdidload is called we are getting the contacts for the first time, then we register method for the appWillEnterForeground.
- So, as requested, when you leave the app and maybe add contact and then go back to the app, appWillEnterForeground will be called (not viewDidLoad)

Reply:
@ognjenns I had in mind viewWillAppear, which is always called (both when starting the App for first time and when resuming the App), but you introduced new callback with UIApplicationWillEnterForegroundNotification which I guess isn't called when starting App for first time, so we 're good